app.js

```diff
@@ -1,18 +1,38 @@
 'use strict';
 
+var userName = 0;
+
+function getUserName() {
+  userName = prompt('Thank you for participating in our study! Please enter your name');
+  console.log(userName);
+  if(userName === '') {
+    userName = 'AAAHHHH!!!!!';
+    console.log('NO NAME GIVEN');
+  }
+}
+
 // array to store all Junk instances
 Junk.allJunk = [];
-var totalClicks = parseInt(0);
-var studyOutput = document.getElementById('studyOutput');
-var flag = false;
+
+//keep track of all clicks
+Junk.totalClicks = 0;
+
+//track previously displayed junk
+Junk.lastShown = [];
```

Comment:
Since these are properties on the `Junk` constructor, you should set them after `Junk` has been declared. It will not break if you do not, but this keeps your code better organized.